Feature: responsiveness

paulgoetze opened the pull request:
Resolves #26.

paulgoetze committed:
Use F key for entering fullscreen mode

This moves the time travel key to "T"

paulgoetze committed:
Add inbetweens for walking cat

paulgoetze committed:
Make game resizable and adjust object speeds

paulgoetze committed:
Allow swiping up for making the cat jump

paulgoetze committed:
Ensure there is always an initial mouse

paulgoetze committed:
Resize backgrounds and animals

paulgoetze committed:
Simplify ground handling, collision and resizing

Instead of colliding the moving objects with the ground, we just collide
with the window bounds and set offsets.

paulgoetze committed:
Resize obstacles and simplify position handling

paulgoetze committed:
Move reference configs to reference config object

paulgoetze committed:
Resize time text

paulgoetze committed:
Add safe-destroy utils function for destroying phaser objects

paulgoetze committed:
Resize batteries

kennethlarsen reviewed (approved):
👍 💪 ✨ Great work!!

The game has been totally unuseable on the screen of my new laptop so hopefully it will be better now.

paulgoetze committed:
Add makeDraggable util for dragging & dropping groups

paulgoetze committed:
Allow charging and traveling by drag & drop

For charging drag and drop the main battery over the battery pack.
For traveling drag and drop the full battery pack over the cat.

paulgoetze committed:
Resize title state

paulgoetze committed:
Resize intro state

paulgoetze committed:
Allow toggling fullscreen mode for title & intro states

This DRYs toggling the fullscreen mode into a util and uses this util in
the main state as well.

paulgoetze commented:
@kennethlarsen I’ll just make another release with the responsiveness stuff. This should make it playable on all devices.
